Views and controls documentation (#422)
compnerd/swift-win32: Documentation/ViewAndControls.md

## Views and controls

Swift-Win32 provides visual components such as labels and buttons.

![View and controls](./Images/views-and-controls.png)

> The visual component library API is inspired by [UIKit](https://developer.apple.com/documentation/uikit/views_and_controls) but their actual implementation is provided by [win32 windows controls](https://docs.microsoft.com/en-us/windows/win32/controls/common-control-window-classes). For example: the `Label` has a similar API than UIKit's `UILabel` and is drawn using the `WC_STATIC` win32 control called.

Reviewer: I believe that the `Windows Common Controls` is the proper name for this subsystem.  I don't think that `win32` is a good idea to use - it can be confusing - do you mean the win32 subsystem?  Do you mean the win32 apis?  Or do you mean 32-bit windows?  The term is somewhat context sensitive, and it might be better to be more precise in general.

Author: Thanks for the feedback. I updated the text accordingly.

## Views and controls

Swift-Win32 provides visual components such as labels and buttons.

![View and controls](./Images/views-and-controls.png)

The visual component library API is inspired by [UIKit](https://developer.apple.com/documentation/uikit/views_and_controls) but its actual implementation is provided by the [Windows Common Controls](https://docs.microsoft.com/en-us/windows/win32/controls/common-control-window-classes) of the win32 API.

> For example: the `Label` has a similar API than UIKit's `UILabel` and is drawn using the `WC_STATIC` Windows control.

Reviewer: This is not grammatically correct.  The `WC_STATIC` is not a Windows control per se.  It is the name of a window class.

```suggestion
For example: the `Label` has an API similar to UIKit's `UILabel` and is drawn using the `WC_STATIC` Windows window class.
```

### Views and controls catalog

| Control      | Windows control    | Description                                              |
| ------------ | ------------------ | -------------------------------------------------------- |
| Label        | WC_STATIC          | A view that displays text                                |
| TextView     | MSFTEDIT_CLASS     | A view that shows scrollable lines of text               |
| ImageView    | WC_STATIC          | A view that display and image                            |
| ProgressView | PROGRESS_CLASS     | A view that renders a progress bar                       |
| Button       | WC_BUTTON          | A control that executes code in response to clicks       |
| DatePicker   | DATETIMEPICK_CLASS | A control that lets the user choose a date               |
| TableView    | WC_LISTBOX         | A view that shows a list of items                        |
| TextField    | MSFTEDIT_CLASS     | A control that lets the user enter text                  |
| Slider       | TRACKBAR_CLASS     | A control that permits to selects a value in a range     |
| Stepper      | UPDOWN_CLASS       | A control that allows to increment and decrement a value |

> | Swicth       | WC_BUTTON          | A control that offers two choices                        |

Reviewer: One thing which is still unclear is who this documentation is useful for.  This doesn't seem material to users of Swift/Win32, so perhaps we should have a separate documentation group for Swift/Win32 developers?

Author: In my mind, I was targeting users of swift/win32. I agree that the second column in not necessary because I already mentioned that the Windows common controls are used and that's enough to know as a user of the library.
